platform-sdk/docs/proposals/state-lifecycle/proposal.md

```diff
@@ -33,66 +33,76 @@ These methods correspond to the state lifecycle as follows:
+  - Keeping the `freeze time` parameter up-to-date.
+  - Providing state for signing.
+  - Initializing the state by extracting it from the signed state.
 
 These two sets of responsibilities should be separated. The `swirlds-state-impl` should not have any details related to the platform. However, it should provide all necessary levers for the platform to interact with the state and its lifecycle.
 
-Therefore, `swirlds-state-api` needs a `StateManager` interface with an implementation `StateManagerImpl` in `swirlds-state-impl`  that will have the following responsibilities:
-- Create an empty state
-- Keeping track of the references to the latest immutable state and the latest mutable state.
-- Tracking states that existed before the latest immutable state. These should be available for lookup by their round number.
-- Restricts mutability to a single state. That is, only one state object should be mutable at any given time.
-- Evicting old states from memory that are no longer needed. `evictState` method *must* fail if the state is still mutable.
-- Loading snapshots from the disk.
+Therefore, `swirlds-state-api` needs a `StateLifecycleManager` interface with an implementation `StateLifecycleManagerImpl` in `swirlds-state-impl`  that will have the following responsibilities:
+- Take initial state as a starting point. It can be a genesis state or a state loaded from a snapshot.
+- Keep track of the references to the latest immutable state and the latest mutable state.
+- Restrict mutability to a single state. That is, only one state object should be mutable at any given time.
+
+The interface will look something like this:
+
+```java
+public interface StateLifecycleManager {
+    void setInitialState(State initialState);
+
+    State getLatestImmutableState();
 
-Additionally, there should be a configuration parameter that limits the number of states kept in memory. When the limit is reached, the oldest state should be evicted.
-Currently, we rely on the implicit garbage collection. We have a reference count mechanism to achieve this.
-We want to get rid of this mechanism and rely on the explicit eviction mechanism.
+    State getMutableState();
+}
+```
+
+**Imporatant**: It's up to the client code to make sure that the state is released once it's no longer needed.
+Without this, the state will not be evicted from memory and eventually the application will run out of memory.
 
 The following diagram illustrates an example of the states in memory throughout their lifecycles:
 
-[![Multistate management](multi-state-management.png)](multi-state-management.png)
+[![Multistate management](multi-states.svg)](multi-states.svg)
+
+Also, `swirlds-state-api` needs a `SnapshotManager` interface with an implementation `SnapshotManagerImpl` in `swirlds-state-impl` that will have the following responsibilities:
+- Load snapshots from the disk. A state loaded from a snapshot must be immutable and available for look up by the round number.
```

Comment:
```suggestion
- Load snapshots from the disk. A state loaded from a snapshot is immutable and available for look up by the round number.
```
The method to get a state by round number was removed which I think is correct. This verbage needs to be updated to say that the snapshot loaded from disk will be fast copied. The immutable version loaded from disk will be available via the call to `getLatestImmutableState()` and the mutable copy will be available via `getMutableState()`